build-system/tasks/helpers.js

```diff
@@ -196,7 +196,7 @@ function compile(watch, shouldMinify) {
     ),
   ];
 
-  if (!argv.single_pass && (!watch || argv.with_shadow)) {
+  if (!argv.single_pass) {
```

Comment:
Does shadow not compile in single pass mode? If it does, you can remove this if block as well.

/cc @erwinmombay